app/uk/gov/hmrc/digitalservicestax/journeys/ReturnJourney.scala

```diff
@@ -47,41 +50,60 @@ object ReturnJourney {
 
     def askAlternativeCharge(applicableActivities: Set[Activity]): F[Map[Activity, Percent]] = {
 
+      def askActivityReduced(actType: Activity): F[Percent] =
+        { ask[Percent](s"report-$actType-operating-margin") emptyUnless
+          ask[Boolean](s"report-$actType-loss")}
+
       def askActivity(actType: Activity): F[Option[Percent]] = 
-      { ask[Percent](s"$actType-margin") emptyUnless
-        ask[Boolean](s"$actType-loss").map { x => !x }} when
-      ask[Boolean](s"$actType-applying")
+        { ask[Percent](s"report-$actType-operating-margin") emptyUnless
+          ask[Boolean](s"report-$actType-loss").map { x => !x }} when
+        ask[Boolean](s"report-$actType-alternative-charge")
       
       val allEntries: List[F[(Activity, Option[Percent])]] =
-        applicableActivities.toList.map{ actType =>
-          askActivity(actType).map{ (actType, _)}
+        if(applicableActivities.size == 1) {
```

Comment:
consider - 

```
applicableActivities.toList match { 
   case actType :: Nil => askActivityReduced(actType).map { ... } 
   case xs => xs.map { actType => ... } 
}
```
This saves having to iterate over a list of 1 (it's probably the same performance wise, but should mean a bit less code).

Reply:
Will refactor this in bext branch